src/app/pages/test-report/cat-b/test-report.cat-b.page.ts

```diff
@@ -68,211 +36,56 @@ interface TestReportPageState {
 export class TestReportCatBPage extends TestReportBasePageComponent implements OnInit {
 
   pageState: TestReportPageState;
-  subscription: Subscription;
-  competencies = Competencies;
-  legalRequirements = LegalRequirements;
-  eta = ExaminerActions;
-  displayOverlay: boolean;
-
-  isRemoveFaultMode: boolean = false;
-  isSeriousMode: boolean = false;
-  isDangerousMode: boolean = false;
-  manoeuvresCompleted: boolean = false;
-  isTestReportValid: boolean = false;
-  isEtaValid: boolean = true;
-
-  modal: HTMLIonModalElement;
-  missingLegalRequirements: legalRequirementsLabels[] = [];
 
   constructor(
     platform: Platform,
     authenticationProvider: AuthenticationProvider,
     router: Router,
     store$: Store<StoreModel>,
-    private modalController: ModalController,
-    public testReportValidatorProvider: TestReportValidatorProvider,
-    public screenOrientation: ScreenOrientation,
-    public insomnia: Insomnia,
-    public statusBar: StatusBar,
+    modalController: ModalController,
+    testReportValidatorProvider: TestReportValidatorProvider,
+    screenOrientation: ScreenOrientation,
+    insomnia: Insomnia,
+    statusBar: StatusBar,
   ) {
-    super(platform, authenticationProvider, router, store$);
+    super(
+      platform,
+      authenticationProvider,
+      router,
+      store$,
+      modalController,
+      testReportValidatorProvider,
+      screenOrientation,
+      insomnia,
+      statusBar,
+    );
     this.displayOverlay = false;
   }
 
-  getCallback(): OverlayCallback {
-    return {
-      callbackMethod: () => {
-        this.toggleReportOverlay();
-      },
-    };
-  }
-
   ngOnInit(): void {
     super.onInitialisation();
-    // this.pageState = {
-    //   ...this.commonPageState,
-    // };
-
-    const currentTest$ = this.store$.pipe(
-      select(getTests),
-      select(getCurrentTest),
-    );
 
     this.pageState = {
-      candidateUntitledName$: currentTest$.pipe(
-        select(getJournalData),
-        select(getCandidate),
-        select(getUntitledCandidateName),
-      ),
-      isRemoveFaultMode$: this.store$.pipe(
-        select(getTestReportState),
-        select(isRemoveFaultMode),
-      ),
-      isSeriousMode$: this.store$.pipe(
-        select(getTestReportState),
-        select(isSeriousMode),
-      ),
-      isDangerousMode$: this.store$.pipe(
-        select(getTestReportState),
-        select(isDangerousMode),
-      ),
-      manoeuvres$: currentTest$.pipe(
-        select(getTestData),
-        select(hasManoeuvreBeenCompletedCatB),
-      ),
-      testData$: currentTest$.pipe(
-        select(getTestData),
-      ),
-      testRequirements$: currentTest$.pipe(
-        select(getTestData),
-        select(getTestRequirementsCatB),
-      ),
+      ...this.commonPageState,
+      testData$: this.commonPageState.testData$ as Observable<CatBUniqueTypes.TestData>,
```

Comment:
any reason we need to do this, Is it just for the ability to cast the data?

Reply:
as discussed